root.go

```diff
@@ -27,28 +27,51 @@ func (r MerkleRoot) MustSum() uint64 {
 // Sum returns the uint64 sum of the merkle root, it checks the length of the
 // merkle root and if it is no the same as the size of the SMST's expected
 // root hash it will return an error.
-func (r MerkleRoot) Sum() (uint64, error) {
-	if len(r)%SmtRootSizeBytes == 0 {
-		return 0, fmt.Errorf("root#sum: not a merkle sum trie")
+func (r MerkleSumRoot) Sum() (uint64, error) {
+	if len(r) != SmstRootSizeBytes {
+		return 0, fmt.Errorf("MerkleSumRoot#Sum: not a merkle sum trie")
 	}
 
-	firstSumByteIdx, firstCountByteIdx := getFirstMetaByteIdx([]byte(r))
+	return getSum(r), nil
+}
 
-	var sumBz [sumSizeBytes]byte
-	copy(sumBz[:], []byte(r)[firstSumByteIdx:firstCountByteIdx])
-	return binary.BigEndian.Uint64(sumBz[:]), nil
+// MustCount returns the uint64 count of the merkle root, a cryptographically secure
+// count of the number of non-empty leafs in the tree. It panics if the root hash length
+// does not match that of the SMST hasher.
+func (r MerkleSumRoot) MustCount() uint64 {
+	count, err := r.Count()
+	if err != nil {
+		panic(err)
+	}
+
+	return count
 }
 
 // Count returns the uint64 count of the merkle root, a cryptographically secure
-// count of the number of non-empty leafs in the tree.
-func (r MerkleRoot) Count() uint64 {
-	if len(r)%SmtRootSizeBytes == 0 {
-		panic("root#sum: not a merkle sum trie")
+// count of the number of non-empty leafs in the tree. It returns an error if the root hash length
```

Comment:
```suggestion
// Count of the number of non-empty leafs in the tree. It returns an error if the root hash length
```

Reply:
This is not the beginning of the sentence, the first line is not part of the diff.